libsplinter/src/auth/oauth/builder/github.rs

```diff
@@ -0,0 +1,72 @@
+// Copyright 2018-2020 Cargill Incorporated
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use crate::auth::oauth::{builder::OAuthClientBuilder, error::OAuthClientBuildError, OAuthClient};
+use crate::auth::rest_api::identity::github::GithubUserIdentityProvider;
+
+/// Builds a new `OAuthClient` with GitHub's authorization and token URLs.
+pub struct GithubOAuthClientBuilder {
+    inner: OAuthClientBuilder,
+}
+
+impl GithubOAuthClientBuilder {
+    /// Constructs a Github OAuthClient builder.
+    pub fn new() -> Self {
+        Self {
+            inner: OAuthClientBuilder {
```

Comment:
This is probably technically fine, but there is no reason to reach into the internals of OAuthClientBuilder from GithubOAuthClientBuilder since this could just use with_* statements in its API.

Reply:
fixed